collections/soldiers.js

```diff
@@ -0,0 +1,36 @@
+const MongoClient = require('mongodb').MongoClient;
+const assert = require('assert');
+const configs = require('../configFile');
+
+// Connection URL
+const url = configs.dbUrl;
+
+// Database Name
+const dbName = configs.dbName;
+
+// Use connect method to connect to the server
+function newSoldier(db, data, callback) {
+  if (!(/[0-9]{7}/.test(data["id"]))) {
+    callback("The soldier json that was sent is didnt had a legal id", null, 400);
+  } else {
+    let fields = ["id", "name", "rank", "limitations"];
+    let areFieldsLegal = true;
```

Comment:
don't use flags, this isn't C, you can just stop and return once there is a problem, or throw an error